publishing-request/src/main/java/no/unit/nva/publication/publishingrequest/GeneralSupportRequestDto.java

```diff
@@ -0,0 +1,147 @@
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.annotation.JsonTypeInfo.Id;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import java.net.URI;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+import no.unit.nva.identifiers.SortableIdentifier;
+import no.unit.nva.publication.model.MessageDto;
+import no.unit.nva.publication.model.business.GeneralSupportRequest;
+import no.unit.nva.publication.model.business.TicketEntry;
+import no.unit.nva.publication.model.business.TicketStatus;
+import nva.commons.core.JacocoGenerated;
+
+@JsonTypeInfo(use = Id.NAME, property = "type")
+@JsonTypeName(GeneralSupportRequestDto.TYPE)
+public class GeneralSupportRequestDto extends TicketDto {
+    
+    public static final String TYPE = "GeneralSupportCase";
+    public static final String STATUS_FIELD = "status";
+    public static final String CREATED_DATE_FIELD = "createdDate";
+    public static final String MODIFIED_DATE_FIELD = "modifiedDate";
+    public static final String VERSION_FIELD = "version";
+    public static final String IDENTIFIER_FIELD = "identifier";
+    public static final String PUBLICATION_ID_FIELD = "publicationId";
+    public static final String ID_FIELD = "id";
+    @JsonProperty(ID_FIELD)
+    private final URI id;
+    @JsonProperty(STATUS_FIELD)
+    private final TicketStatus status;
+    @JsonProperty(CREATED_DATE_FIELD)
+    private final Instant createdDate;
+    @JsonProperty(MODIFIED_DATE_FIELD)
+    private final Instant modifiedDate;
+    @JsonProperty(VERSION_FIELD)
+    private final UUID version;
+    @JsonProperty(IDENTIFIER_FIELD)
+    private final SortableIdentifier identifier;
+    @JsonProperty(PUBLICATION_ID_FIELD)
+    private final URI publicationId;
+    @JsonProperty(MESSAGES_FIELD)
+    private final List<MessageDto> messages;
+    
+    public GeneralSupportRequestDto(@JsonProperty(STATUS_FIELD) TicketStatus status,
+                                    @JsonProperty(CREATED_DATE_FIELD) Instant createdDate,
+                                    @JsonProperty(MODIFIED_DATE_FIELD) Instant modifiedDate,
+                                    @JsonProperty(VERSION_FIELD) UUID version,
+                                    @JsonProperty(IDENTIFIER_FIELD) SortableIdentifier identifier,
+                                    @JsonProperty(PUBLICATION_ID_FIELD) URI publicationId,
+                                    @JsonProperty(ID_FIELD) URI id,
+                                    @JsonProperty(MESSAGES_FIELD) List<MessageDto> messages) {
+        super();
+        this.status = status;
+        this.createdDate = createdDate;
+        this.modifiedDate = modifiedDate;
+        this.version = version;
+        this.identifier = identifier;
+        this.publicationId = publicationId;
+        this.id = id;
+        this.messages = messages;
```

Comment:
Honestly, it isn't important right now, but I feel the organisation of the signature here doesn't reflect what is important in the creation of these objects. Is there are reason to expose the version and identifier here?

Reply:
version yes, identifier no